epick_moveit_studio/src/compare_epick_object_detection_status.cpp

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2023 PickNik, Inc.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//    * Redistributions of source code must retain the above copyright
+//      notice, this list of conditions and the following disclaimer.
+//
+//    * Redistributions in binary form must reproduce the above copyright
+//      notice, this list of conditions and the following disclaimer in the
+//      documentation and/or other materials provided with the distribution.
+//
+//    * Neither the name of the {copyright_holder} nor the names of its
+//      contributors may be used to endorse or promote products derived from
+//      this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#include <behaviortree_cpp/action_node.h>
+#include <behaviortree_cpp/basic_types.h>
+#include <epick_moveit_studio/compare_epick_object_detection_status.hpp>
+#include <epick_msgs/msg/object_detection_status.hpp>
+
+namespace
+{
+constexpr auto kPortIDValue1 = "value1";
+constexpr auto kPortIDValue2 = "value2";
+}  // namespace
+
+namespace epick_moveit_studio
+{
+CompareEpickObjectDetectionStatus::CompareEpickObjectDetectionStatus(const std::string& name,
+                                                                     const BT::NodeConfiguration& config)
+  : BT::SyncActionNode(name, config)
+{
+}
+
+BT::PortsList CompareEpickObjectDetectionStatus::providedPorts()
+{
+  return BT::PortsList{
+    BT::InputPort<epick_msgs::msg::ObjectDetectionStatus>(kPortIDValue1),
+    BT::InputPort<epick_msgs::msg::ObjectDetectionStatus>(kPortIDValue2),
+  };
+}
+
+BT::NodeStatus CompareEpickObjectDetectionStatus::tick()
+{
+  const auto value1 = getInput<epick_msgs::msg::ObjectDetectionStatus>(kPortIDValue1);
+  const auto value2 = getInput<epick_msgs::msg::ObjectDetectionStatus>(kPortIDValue2);
+
+  if (!value1.has_value() || !value2.has_value())
+  {
+    return BT::NodeStatus::FAILURE;
+  }
```

Comment:
Do we want to log a failure message here, or is it a valid workflow to have this be blank?

Reply:
This is an error state, but it bugs me that I'd need to make an otherwise very simple behavior derive from `SharedResourcesNode` just to emit an error reason. BT.Cpp does print an error message to logs when it fails to read a port's value, so it's not totally silent in this situation anyway.

Reply:
agreed. I didn't realize this one didn't already inherit from SharedResourcesNode. As you were!